src/parser/api/cpp/command.cpp

```diff
@@ -1075,7 +1079,10 @@ void DefineFunctionCommand::invoke(cvc5::Solver* solver, SymbolManager* sm)
     bool global = sm->getGlobalDeclarations();
     cvc5::Term fun =
         solver->defineFun(d_symbol, d_formals, d_sort, d_formula, global);
-    sm->bind(d_symbol, fun, global);
+    if (d_reqBinding)
```

Comment:
I believe you don't have to make the `fun` in this case either.  I think the entire invoke does nothing actually in the case that d_reqBinding is false.